web-ui/src/streaming/import/random-data/generators.ts

```diff
@@ -0,0 +1,1067 @@
+    form_fields: () => [
+      {
+        sm: 2,
+        component: TextField,
+        props: {
+          fullWidth: true,
+          type: 'number',
+          name: FieldNames.MU,
+          label: 'μ'
+        }
+      },
+      {
+        sm: 2,
+        component: TextField,
+        props: { fullWidth: true, type: 'number', name: FieldNames.SIGMA, label: 'σ', inputProps: { step: 0.1 } }
+      }
+    ],
+    validationSchema: ct => {
+      const [min, max] = typeRange(ct)
+      return yup.object().shape({
+        mu: yup.number().min(min).max(max).required('Mu value is required.'),
+        sigma: yup.number().min(min).max(max).required('Sigma is required.')
+      })
+    }
+  },
+  {
+    title: 'Exponential',
+    category: Categories.DISTRIBUTION,
+    generator: (ct, props) => Math.floor(d3.randomExponential(props?.lambda || 0.1)()),
+    form_fields: () => [
+      {
+        sm: 2,
+        component: TextField,
+        props: {
+          fullWidth: true,
+          type: 'number',
+          name: FieldNames.LAMBDA,
+          label: 'λ',
+          inputProps: { step: 0.1 }
+        }
+      }
+    ],
+    validationSchema: () =>
+      yup.object().shape({
+        lambda: yup.number().required('Lambda is required.')
+      })
+  }
+]
+
+const INTEGER_GENERATORS: IRngGenMethod[] = NUMBER_GENERATORS.concat([
+  {
+    title: 'Uniform',
+    category: Categories.DISTRIBUTION,
+    generator: (ct, props) => {
+      const [typeMin, typeMax] = typeRange(ct)
+      const min = Math.ceil(props?.min ?? typeMin)
+      const max = Math.floor(props?.max ?? typeMax)
+      return d3.randomInt(min, max)()
+    },
+    form_fields: () => [
+      {
+        sm: 2,
+        component: TextField,
+        props: {
+          fullWidth: true,
+          type: 'number',
+          name: FieldNames.MIN,
+          label: 'Minimum'
+        }
+      },
+      {
+        sm: 2,
+        component: TextField,
+        props: { fullWidth: true, type: 'number', name: FieldNames.MAX, label: 'Maximum' }
+      }
+    ],
+    validationSchema: ct => {
+      const [min, max] = typeRange(ct)
+      return yup.object().shape({
+        min: yup.number().min(min).max(max),
+        max: yup.number().min(min).max(max)
+      })
+    }
+  }
+])
+
+// Generators for floating point numbers.
+//
+// This 'inherits' all the integer generators.
+const FLOAT_GENERATORS: IRngGenMethod[] = NUMBER_GENERATORS.concat([
+  {
+    title: 'Uniform',
+    category: Categories.DISTRIBUTION,
+    generator: (ct, props) => {
+      if (props?.min && props.max) {
+        // We use d3.randomUniform for generating floating point numbers as long
+        // as the user set a range for the values.
+        const min = props?.min
+        const max = props?.max
+        return d3.randomUniform(min, max)()
+      } else {
+        // Because it's surprisingly difficult to generate floating point
+        // numbers with a max range of -Inf..Inf, we just default to 0..1 for
```

Comment:
what's the point to generate something that can't be represented?

Reply:
I'd be happy with someting that gives me uniform random in the full range that *can* be represented by number... [but it seems more difficult than I wanted to spend time on it](https://stackoverflow.com/questions/28461796/randomint-function-that-can-uniformly-handle-the-full-range-of-min-and-max-safe)